syncd/syncd_saiswitch.cpp

```diff
@@ -1220,9 +1392,27 @@ SaiSwitch::SaiSwitch(
 
     helperInternalOids();
 
-    helperCheckLaneMap();
+    m_port_rid_list = saiGetPortList();
```

Comment:
this won't work with the port break out mode. This one is executed before you remove/add port, so the port list needs to be updated once the break out logic is executed.

Reply:
In this case I will revert the changes and syncd will read the port and queue list each time on the counter’s update.
The reason for that revert is we don’t have a notification in the syncd when portsyncd in swss removes/creates the ports.

Reply:
can you hook with remove_port api to remove the port from the port_list?

Reply:
I think the best approach is to do something simliar to pfc watchdog, let swss to tell syncd the ports and counters to query periodically.

Reply:
First of all, thank you for your suggestions regarding to syncd port/queue list implementation. 

From my point of view, the best approach is to update the port/queue list on remote_port api notification. 
I’ve checked the port break out behavior. We get the event from database when port is created/removed and could add a new case:
If port or queue SAI object is created/removed, we should update the port and queue list.
Also I have an following open question to this suggestion:
Should we update the port/queue list on a set operation?

In other hand, we can update the port/queue counters in similar way as it’s done in pfc watchdog. 
We can add/remove the port’s oid  to DB when port add/remove api in swss is called and update it in syncd_flex_counter component. 
The main disadvantage of this approach is that pfc watchdog has a very small updating interval = 100 msec. 
In this case, we need to do one from below things:
•	duplicate the implementation of syncd_flex_counter component for port/queue updating 
•	refactor the syncd_flex_counter component and make the updating interval configurable for different oid


Which of above approaches should be implemented for optimizing the port and queue counters update?